sql/database.go

```diff
@@ -582,3 +636,15 @@ func LoadBlob(db Executor, cmd string, id []byte, blob *Blob) error {
 func IsNull(stmt *Statement, col int) bool {
 	return stmt.ColumnType(col) == sqlite.SQLITE_NULL
 }
+
+// StateDatabase is a Database used for Spacemesh state.
+type StateDatabase interface {
+	Database
+	IsStateDatabase() bool
```

Comment:
NIT: It doesn't need to return anything if its purpose is to simply distinguish local vs state
```suggestion
	IsStateDatabase()
```

Reply:
can it maybe even be a private method?

Reply:
Removed return value, but can't make these methods private as the interfaces are implemented in `sql/statesql` and `sql/localsql` packages and unexported interface methods can't be implemented in the packages other than the package where the interface is declared.